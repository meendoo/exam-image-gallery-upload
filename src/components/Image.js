import React, { Component } from "react";
import styles from "./image.module.scss";
import CaptionInput from "./CaptionInput";

export default class Image extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isCaptionEditable: false
    };
  }

  toggleInputCaption = (e, state) => {
    e.stopPropagation();
    this.setState({
      isCaptionEditable: state
    });
  };

  render() {
    const { imageId, updateImageCaption, className, url, caption } = this.props;
    const { isCaptionEditable } = this.state;
    return (
      <figure className={`${styles.figure} ${className}`}>
        <div
          style={{ backgroundImage: `url('${url}')` }}
          className={`${styles.imgDiv} ${className}`}
        />
        <figcaption className={styles.figcaption} onClick={e => this.toggleInputCaption(e, true)}>
          {!isCaptionEditable ? (
            <p>{String(caption).substr(0, 250)}</p>
          ) : (
            <CaptionInput
              caption={caption}
              imageId={imageId}
              hideInput={e => this.toggleInputCaption(e, false)}
              updateImageCaption={updateImageCaption}
            />
          )}
        </figcaption>
      </figure>
    );
  }
}
