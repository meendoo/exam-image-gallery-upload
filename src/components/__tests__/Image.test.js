import React from 'react'
import renderer from 'react-test-renderer'
import Image from '../Image'

test('Image component snapshot', () => {
    const tree = renderer.create(<Image/>).toJSON();
    expect(tree).toMatchSnapshot();
})
