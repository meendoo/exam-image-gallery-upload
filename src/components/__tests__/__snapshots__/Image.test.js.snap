// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component snapshot 1`] = `
<figure
  className="figure undefined"
>
  <div
    className="imgDiv undefined"
    style={
      Object {
        "backgroundImage": "url('undefined')",
      }
    }
  />
</figure>
`;
